packages/strapi/lib/core-api/queries/bookshelf.js

```diff
@@ -198,7 +198,7 @@ const buildSearchQuery = (qb, model, params) => {
           : `to_tsvector('${attribute}')`
       );
 
-      qb.orWhereRaw(`${searchQuery.join(' || ')} @@ to_tsquery(?)`, query);
+      qb.orWhereRaw(`${searchQuery.join(' || ')} @@ plainto_tsquery(?)`, query);
```

Comment:
Ok sorry I wasn't clear enough remove the replace is ok but the queries need some refactoring to pass those strings as bindings not directly in a template string. otherwise we will have sql injection issues

Reply:
Right now the mysql and searchQuery in pg are vulnerable

Reply:
They're not, as we're using prepared statements to build the query, which then gets escaped by the underlying library. I tested it out with common injection strings and they all got escaped.

I'll attach some examples of what I've tested using the `testApp` generated by the e2e tests.

Given an input of `' or 1=1` the query produced looks like this:

```sql
select * from "products" where to_tsvector(name) || to_tsvector(description) @@ plainto_tsquery(''' or 1=1')
```

As you can see the quotes get escaped by the query builder not allowing for any SQL injection attacks.

The only non-parametrized part of the query is where we're putting `searchQuery` in the template literal, but that doesn't contain any outside input and is derived from the model itself.